Update the hls4ml documentation

aperloff opened the pull request:
Update the hls4ml documentation to point to known recent tutorials and add some boilerplate material. This documentation doesn't duplicate any of the hls4ml documentation or API. It simply points the users to the relevant, primary documentation source. The list of tutorials will be added to as we are made aware of new ones.

@thaarres @thesps @jmduarte @jngadiub @nhanvtran

thesps commented:
Looks nice to me, thanks for putting it together. I noticed a typo that I've left an inline comment for.

To the list of tutorials you could add 4th IML workshop, Oct 19 2020 ([link](https://indico.cern.ch/event/852553/timetable/#b-394792-hls4ml-tutorial-hls4m)) and [30th International Conference on Field Programmable Logic and Applications (FPL)](https://www.fpl2020.org/) ([link](https://www.fpl2020.org/attending/program)), Sep 4 2020

aperloff commented:
> Looks nice to me, thanks for putting it together. I noticed a typo that I've left an inline comment for.
> 
> To the list of tutorials you could add 4th IML workshop, Oct 19 2020 ([link](https://indico.cern.ch/event/852553/timetable/#b-394792-hls4ml-tutorial-hls4m)) and [30th International Conference on Field Programmable Logic and Applications (FPL)](https://www.fpl2020.org/) ([link](https://www.fpl2020.org/attending/program)), Sep 4 2020

@thesps Thank you for the comments. I've fixed the typo and added the tutorials you pointed out.
